docs: add instructions for activating Olares device using Olares CLI (#3060)
beclab/Olares: docs/manual/best-practices/activate-olares-using-cli.md

---
outline: [2, 3]
description: Technical guide for installing and activating an Olares device using Olares CLI.
---

# Activate an Olares device using the Olares CLI

This tutorial walks you through activating an Olares device (e.g., Olares One) using the Olares CLI tool. The process assumes the device is freshly unboxed and has not been installed or activated.

> ## Objectives

Reviewer: ```suggestion
## Learning objectives
```

In this tutorial, you will learn how to:

- Download and extract the Olares CLI tool.
- Install the Olares OS on a new device.
- Retrieve a Fast Reverse Proxy (FRP) host for remote access.
- Run the activation command to configure your device.

## Prerequisites

- **Connection to device**: You have logged into the Olares device physically or via SSH.
- **Root privileges**: Ensure you have root privileges. All installation and activation commands require `sudo` or root access on the device.
- **LarePass preparation**: You have registered an Olares ID using the LarePass app.

    ![Fast creation](/images/manual/get-started/create-olares-id.png)

## Step 1: Download and extract the CLI tool

1. Download the Olares CLI package.

>     ```bash

Reviewer: Should we add the network requirement in ##Prerequisites?

---
outline: [2, 3]
description: Technical guide for installing and activating an Olares device using Olares CLI.
---

# Activate an Olares device using the Olares CLI

This tutorial walks you through activating an Olares device (e.g., Olares One) using the Olares CLI tool. The process assumes the device is freshly unboxed and has not been installed or activated.

## Learning objectives

In this tutorial, you will learn how to:

> - Download and extract the Olares CLI tool.
> - Install the Olares OS on a new device.
> - Retrieve a Fast Reverse Proxy (FRP) host for remote access.
> - Run the activation command to configure your device.
> 
> ## Prerequisites
> 
> - **Connection to device**: You have logged into the Olares device physically or via SSH.
> - **Network connectivity**: The device must have access to the internet to download packages, query FRP servers, and complete activation.
> - **Root privileges**: Ensure you have root privileges. All installation and activation commands require `sudo` or root access on the device.
> - **LarePass preparation**: You have registered an Olares ID using the LarePass app.
> 

Reviewer: ```suggestion
## Prerequisites

  Before you begin, make sure the following requirements are met:

  - You can access the Olares device directly with a keyboard and monitor, or via
  SSH.
  - The device has internet access to download packages, query FRP servers, and
  complete activation.
  - You can run commands as the root user, or prepend commands with `sudo`.
  - You have created an Olares ID using the LarePass app, and have [backed up your
   12-word mnemonic phrase](../larepass/back-up-mnemonics.md).

```

    ![Fast creation](/images/manual/get-started/create-olares-id.png)

## Step 1: Download and extract the CLI tool

1. Download the Olares CLI package.

    ```bash

>     curl -sSOL https://cdn.olares.com/common/olares-cli-amd64.8cbdc32.tar.gz

Reviewer: Curious: Does it mean we should update this link every time there’s a new release of olares-cli?

Author: Confirmed with dev: As long as there are no bugs, the same link will work.

    ```

2. Extract the downloaded file.

    ```bash
    tar xzf olares-cli-amd64.8cbdc32.tar.gz
    ```

## Step 2: Retrieve the FRP list

Find an available FRP host to enable remote access to your device.

1. Run the following command. Replace `{olares-id}` with your registered Olares ID.

    ```bash
    olares-cli wizard frp {olares-id}
    ```

    **Example:**

    ```bash
    olares-cli wizard frp alice123@olares.com
    ```

2. Select a host address from the output list and save it for the activation step.

> ## Step 3: Install Olares OS

Reviewer: ```suggestion
## Step 3: Install Olares
```

:::info For Olares One hardware
A fresh Olares One device is shipped in an uninstalled state. You must run the installation command to set up Olares first before you attempt activation.
:::

1. Run the install command.

    ```bash
    sudo olares-cli install
    ```

2. Wait for the installation process to finish. The terminal outputs a local gateway address and a default password. Save these details for the activation step.

    **Example:**
    - **Wizard URL**: The local gateway address, such as `http://192.168.50.123:30180`.
    - **Password**: The default Olares login password.

    ![Wizard URL](/images/manual/get-started/wizard-url-and-login-password.png)

## Step 4: Activate Olares

Run the activation command to configure and secure your device. This process connects your Olares ID to the device and configures network tunneling and credentials.

1. Prepare your activation command based on the following required parameters:

    | Parameter | Description |
    |:----------|:------------|

>     | `olares-id` | The unique identifier within the Olares ecosystem.<br>Find it in the LarePass app after registration. |

Reviewer: ```suggestion
    | `olares-id` | The Olares ID you created in LarePass, <br>for example
  `alice123@olares.com`. |
```

>     | `mnemonic` | Your user backup phrase from the LarePass app. |

Reviewer: ```suggestion
    | `mnemonic` | The 12-word mnemonic phrase of your Olares ID. |
```

    | `password` | The default Olares login password generated in Step 3. |
    | `reset-password` | Specify a new login password for Olares. |
    | `authurl` | Enter the `wizard-url` generated in Step 3. |
    | `vault` | Enter the `wizard-url` and append `/server`.  |
    | `bfl` | Enter the `wizard-url` generated in Step 3. |

>     | `host` | Specify the FRP host address selected in Step 2. |

Reviewer: ```suggestion
    | `host` | Enter the FRP host address specified in Step 2. |
```

    **Example:**
    - **Wizard URL**: The local gateway address, such as `http://192.168.50.123:30180`.
    - **Password**: The default Olares login password.

    ![Wizard URL](/images/manual/get-started/wizard-url-and-login-password.png)

## Step 4: Activate Olares

Run the activation command to configure and secure your device. This process connects your Olares ID to the device and configures network tunneling and credentials.

1. Prepare your activation command based on the following required parameters:

    | Parameter | Description |
    |:----------|:------------|
    | `olares-id` | The Olares ID you created in LarePass, <br>for example
  `alice123@olares.com`. |
    | `mnemonic` | The 12-word mnemonic phrase of your Olares ID. |
    | `password` | The default Olares login password generated in Step 3. |
    | `reset-password` | Specify a new login password for Olares. |
    | `authurl` | Enter the `wizard-url` generated in Step 3. |
    | `vault` | Enter the `wizard-url` and append `/server`.  |
    | `bfl` | Enter the `wizard-url` generated in Step 3. |
    | `host` | Enter the FRP host address specified in Step 2. |

>     | `enable-tunnel` | Enter `true` to activate using tunnel mode. |

Reviewer: ```suggestion
    | `password` | The default Olares login password from Step 3. |
    | `reset-password` | A new login password to replace the default one. |
    | `authurl` | The Wizard URL from Step 3. |
    | `vault` | The Wizard URL from Step 3, followed by `/server`.  |
    | `bfl` | The Wizard URL from Step 3. |
    | `host` | The FRP host address from Step 2. |
    | `enable-tunnel` | Set to `true` to enable tunnel mode. |
```

To align the description
